src/contracts/Calculator.sol

```diff
@@ -1194,47 +1243,58 @@ contract Calculator is OwnableUpgradeable, ICalculator {
   /// @param _averagePrice The average price of the position.
   /// @return isProfit A boolean value indicating whether the position is profitable or not.
   /// @return delta The Profit between the average price and the fixed price, adjusted for the size of the order.
-  function _getDelta(
-    uint256 _size,
-    bool _isLong,
-    uint256 _markPrice,
-    uint256 _averagePrice,
-    uint256 _lastIncreaseTimestamp,
-    uint256 _marketIndex,
-    bool _useMinProfitDuration
-  ) internal view returns (bool, uint256) {
+  struct GetDeltaVars2 {
+    address subAccount;
+    uint256 size;
+    bool isLong;
+    uint256 markPrice;
+    uint256 averagePrice;
+    uint256 lastIncreaseTimestamp;
+    uint256 marketIndex;
+    bool useMinProfitDuration;
+    uint256 priceDelta;
+    uint256 delta;
+    bool isProfit;
+  }
+
+  function _getDelta(GetDeltaVars2 memory vars) internal view returns (bool, uint256) {
     // Check for invalid input: averagePrice cannot be zero.
-    if (_averagePrice == 0) return (false, 0);
+    if (vars.averagePrice == 0) return (false, 0);
 
     // Calculate the difference between the average price and the fixed price.
-    uint256 priceDelta;
+    vars.priceDelta;
     unchecked {
-      priceDelta = _averagePrice > _markPrice ? _averagePrice - _markPrice : _markPrice - _averagePrice;
+      vars.priceDelta = vars.averagePrice > vars.markPrice
+        ? vars.averagePrice - vars.markPrice
+        : vars.markPrice - vars.averagePrice;
     }
 
     // Calculate the delta, adjusted for the size of the order.
-    uint256 delta = (_size * priceDelta) / _averagePrice;
+    vars.delta = (vars.size * vars.priceDelta) / vars.averagePrice;
 
     // Determine if the position is profitable or not based on the averagePrice and the mark price.
-    bool isProfit;
-    if (_isLong) {
-      isProfit = _markPrice > _averagePrice;
+    vars.isProfit;
+    if (vars.isLong) {
+      vars.isProfit = vars.markPrice > vars.averagePrice;
     } else {
-      isProfit = _markPrice < _averagePrice;
+      vars.isProfit = vars.markPrice < vars.averagePrice;
     }
 
     // In case of profit, we need to check the current timestamp against minProfitDuration
     // in order to prevent front-run attack, or price manipulation.
     // Check `isProfit` first, to save SLOAD in loss case.
-    if (isProfit && _useMinProfitDuration) {
-      uint256 minProfitDuration = ConfigStorage(configStorage).minProfitDurations(_marketIndex);
-      if (block.timestamp < _lastIncreaseTimestamp + minProfitDuration) {
-        return (isProfit, 0);
+    if (vars.isProfit && vars.useMinProfitDuration) {
+      uint256 minProfitDuration = ConfigStorage(configStorage).getStepMinProfitDuration(
+        vars.marketIndex,
+        PerpStorage(perpStorage).lastIncreaseSizeByPositionId(HMXLib.getPositionId(vars.subAccount, vars.marketIndex))
```

Comment:
if we have position last increase position we can just use
```
PerpStorage.positions(HMXLib.getPositionId(vars.subAccount, vars.marketIndex)).lastIncreasedSize
```
??

Reply:
or we can passed it as vars?

Reply:
Code is wrong. Will fix. Should get `lastIncreaseSize` from position.